data/campaigns/The_Hammer_of_Thursagan/scenarios/12_The_Underlevels.cfg

```diff
@@ -246,7 +247,52 @@
         {VARIABLE rune_one 0}
         {VARIABLE rune_two 0}
     [/event]
-
+    
+    [event]
+        name=recruit
+        first_time_only=no
+        [filter]
+            side=1
+        [/filter]
+        
+        [object]
+            silent=yes
+            take_only_once=no
+            [filter]
+                x,y=$x1,$y1
+            [/filter]
+            [effect]
+                apply_to=vision_costs
+                replace=yes
+                [vision_costs]
+                    impassable=3
+                [/vision_costs]
+            [/effect]   
+        [/object]
+    [/event]
+    [event]
+        name=recall
```

Comment:
event names can be a combination of muliple ones. E.g.
```
name=prerecruit, recall
first_time_only=yes
```
would fire only once, for whatever of the two may happen first.

Here the relatively new `unit placed` event would be a good fit.